plugins/aggregate/aggregate.cpp

```diff
@@ -422,9 +421,12 @@ struct aggregation {
       auto result = std::vector<bool>{};
       result.resize(builder_->num_fields(), false);
       if (!buckets_.empty())
-        for (int column = 0; column < builder_->num_fields(); ++column)
-          result[column] = !builder_->GetField(column)->type()->Equals(
-            buckets_.begin()->second[column]->type);
+        for (int column = 0; column < builder_->num_fields(); ++column) {
+          auto scalar = buckets_.begin()->second[column];
+          result[column]
+            = scalar
+              && !builder_->GetField(column)->type()->Equals(scalar->type);
```

Comment:
This change doesn't make any sense. You can only certainly say that you don't need to cast if the scalars for this column in _all_ buckets are null. This fix just exchanges one bug for another.

Reply:
A better approach would be to change `column_needs_cast[column]` to three possible values: true, false, unknown. If it is unknown the algorithm can try to determine the value on each row until it can be determined.

Reply:
I have changed this behavior accordingly.